### PR TITLE
Fix the discrepancy between nicks and names in pills and mentions

### PR DESCRIPTION
**Pull request by Cadair**
This completely refactors `BaseSlackHandler.replaceUserIdsWithNames` to first check the UserStore for the display name and then fallback on querying slack for the display name. This means it should always be consistent with the nick of any user in the matrix room.

Signed off by: Stuart Mumford <stuart@cadair.com>

### CODE REVIEW

File: lib/BaseSlackHandler.js [hunk 1]
```diff
@@ -69,84 +70,43 @@ BaseSlackHandler.prototype.replaceChannelIdsWithNames = function(message, token)
...
-            if (result === undefined) {
-                return;
-            }
-            let name = user_id;
-            log.info(`${user_id} did ${result.length > 0 ? "not" : ""} an entry`);
-            if (result.length) {
-                // It's possible not to have a displayname set.
-                name = result[0].display_name || result[0].id;
-            }
+        var id = match[0].match(USER_ID_REGEX_FIRST)[1];
```
Comment (Half-Shot) [inline lib/BaseSlackHandler.js]: VAR!

**Comment by Cadair** [comment]
Alright @Half-Shot it's only been a month and a half, but I changed that `var` :rofl:

File: lib/BaseSlackHandler.js [hunk 2]
```diff
@@ -69,84 +70,43 @@ BaseSlackHandler.prototype.replaceChannelIdsWithNames = function(message, token)
...
+        let user_id = main.getUserId(id, team_domain);
+
+        const store = main.getUserStore();
+        const users = await store.select({id: user_id});
+
+        if (users === undefined || !users.length) {
+            log.warn("Mentioned user not in store. Looking up display name from slack.");
+            // if the user is not in the store then we look up the displayname
+            const nullGhost = new slackGhost({main: main});
+            var room = main.getRoomBySlackChannelId(message.channel);
```
Comment (Half-Shot) [inline lib/BaseSlackHandler.js]: VAR :cry:

Comment (Cadair) [reply lib/BaseSlackHandler.js]: done :)

**Comment by Half-Shot** [comment]
Oh, go on then!

**Pull request merged**